server-spi-private/src/main/java/org/keycloak/models/OAuth2DeviceTokenStoreProvider.java

```diff
@@ -72,6 +74,8 @@
      */
     boolean approve(RealmModel realm, String userCode, String userSessionId);
 
+    boolean approve(RealmModel realm, String userCode, String userSessionId, Map<String, String> additionalParams);
```

Comment:
For the simplification purpose, I vote for keep this new method and remove the existing method `boolean approve(RealmModel realm, String userCode, String userSessionId)` from this provider. The existing invocation of the method `approve(realm, userCode, userSessionId)` from the DeviceGrantType can be changed to include null as the last argument. What do you think?